src/box/lua/net_box.lua

```diff
@@ -1213,25 +1320,27 @@ end
 
 function remote_methods:ping(opts)
     check_remote_arg(self, 'ping')
-    return (pcall(self._request, self, 'ping', opts))
+    return (pcall(self._request, self, 'ping', opts, nil, self._stream_id))
 end
 
 function remote_methods:reload_schema()
     check_remote_arg(self, 'reload_schema')
     self:ping()
+    -- delete all invalid spaces in all connection streams
+    collectgarbage()
```

Comment:
You should not need it. You are not using weak refs for spaces anywhere.

Reply:
Fixed